First pass at revising the introduction (#183)
jaybee84/ml-in-rd: content/02.intro.md

## Introduction {.page_break_before}

Rare disease research is increasingly dependent on high-throughput profiling of samples and would greatly benefit from machine learning (ML) analytics. 
Machine learning algorithms are computational methods that can identify patterns in data, and can use information about these patterns to perform tasks (e.g., pick out important data points or predict outcomes when they are not yet known, among other tasks). 
A systematic review of application of ML in rare disease uncovered 211 human data studies that used ML to study 74 different rare diseases over the last 10 years [@doi:10.1186/s13023-020-01424-6]. 
Indeed, ML can be a powerful tool in biomedical research but it does not come without pitfalls [TODO: ref], some of which are magnified in a rare disease context.
In this perspective, we will focus our discussion on considerations for two types of ML in the context of the study of rare diseases: supervised and unsupervised learning.

> _Supervised algorithms_ require training data with specific phenotype labels (e.g., responder vs. non-responder) and learn correlations of features with the phenotype labels to predict the phenotype labels of unseen or new test data (e.g., predicting which patients did or did not respond to treatment).

Reviewer: ```suggestion
_Supervised algorithms_ require training data with specific phenotype labels (e.g., responder vs. non-responder) and learn correlations of features with the phenotype labels to predict the phenotype labels of unseen or new test data (e.g., predicting which new patient would or would not respond to treatment).
```

If the goal of a study is to classify patients with a rare disease into molecular subtypes based on high-throughput profiling, researchers would select a supervised ML algorithm to carry out this task. 
A supervised ML model is of limited utility if it can only accurately predict phenotype labels in the data it was trained on (this is called _overfitting_); instead, it's more beneficial to develop models that _generalize_ or maintain performance when applied to new data that has not yet been "seen" by the model.
In later sections, we'll discuss regularized models, a strategy for reducing overfitting that can be useful for rare disease datasets. 

> _Unsupervised algorithms_ can learn patterns or features from _unlabeled_ training data.

Reviewer: It might be better to tackle the unsupervised first and then move to supervised.

Author: Works for me!

Reviewer: ok.. will merge this one and start a fresh PR :)

Though researchers strive to train useful and informative models, there are challenges inherent to applying ML to rare disease datasets. 
For example, training supervised models requires datasets where the phenotype labels have very little uncertainty (or “label-noise”) [@doi:10.1093/jamia/ocw028] – termed "gold standard" datasets – but rare disease datasets often come with significant label-noise (e.g., _silver standard_ datasets) due to limits in the current understanding of underlying biology and evolving clinical definitions of many rare diseases. 
Label-noise can decrease prediction accuracy and require larger samples sizes during training [@doi:10.1109/tnnls.2013.2292894].
ML methods also benefit from using large datasets, but analyzing high dimensional data from rare diseases datasets that typically contain 20 to 99 samples is challenging [@https://www.fda.gov/media/99546/download; @doi:10.1186/s13023-020-01424-6].
Small datasets lead to a lack of statistical power and magnify the susceptibility of ML methods to misinterpretation and unstable performance.

> While we expect ML in rare disease research to continue increase in popularity, specialized computational methods that can learn patterns from small datasets and can generalize to newly acquired data are required for rare disease applications [@doi:10.1016/j.ebiom.2019.08.027]. 

Reviewer: ```suggestion
While we expect ML in rare disease research to continue to increase in popularity, specialized computational methods that can learn patterns from small datasets and can generalize to newly acquired data are required for rare disease applications [@doi:10.1016/j.ebiom.2019.08.027]. 
```

## Introduction {.page_break_before}

Rare disease research is increasingly dependent on high-throughput profiling of samples and would greatly benefit from machine learning (ML) analytics. 

> Machine learning is TODO: pithy definition

Reviewer 2: ```suggestion
Machine learning algorithms are computational methods that can identify patterns in data, and can use information about these patterns to perform tasks (e.g. pick out important data points or predict outcomes when they are not yet known, among other tasks). 
```

> A systematic review of application of ML in rare disease in the last 10 years uncovered 211 human data studies in 74 different rare diseases [@doi:10.1186/s13023-020-01424-6]. 

Reviewer 2: ```suggestion
A systematic review of application of ML in rare disease in the last 10 years uncovered 211 human data studies that used ML to study 74 different rare diseases [@doi:10.1186/s13023-020-01424-6]. 
```

Indeed, ML can be a powerful tool in biomedical research but it does not come without pitfalls [TODO: ref], some of which are intensified in a rare disease context.
In this perspective, we will focus our discussion on considerations for two types of ML in the context of the study of rare diseases: supervised and unsupervised learning.

> Supervised ML algorithms require training data with specific phenotype labels and learn correlations of features with the phenotype labels to predict the phenotype labels of unseen or new test data.

Reviewer 2: Is it worth describing what we mean by `phenotype labels` in terms a non-ml person would be familiar with? (e.g. `responder vs non-responder to a drug treatment`) in this sentence? I see it in the next but wonder if these could be combined.

> Supervised ML algorithms require training data with specific phenotype labels and learn correlations of features with the phenotype labels to predict the phenotype labels of unseen or new test data.

Reviewer 2: ```suggestion
_Supervised algorithms_ require training data with specific phenotype labels and learn correlations of features with the phenotype labels to predict the phenotype labels of unseen or new test data.
```
This is just a thought, maybe we could italicize or bold terms the first time they show up/when they are defined?

Reviewer 2: I see below that you have done this for some other terms, so maybe we should formalize this/decide to do it for all "newly-introduced" terms? (or maybe we already decided to do it, and I missed it :) )

Predicting which patients did or did not respond to a treatment is a supervised ML task or problem.

> Similarly, if the goal of a study is to classify patients with a rare disease into molecular subtypes based on high-throughput profiling, researchers would select a supervised ML algorithm.

Reviewer 2: ```suggestion
Similarly, if the goal of a study is to classify patients with a rare disease into molecular subtypes based on high-throughput profiling, researchers would select a supervised ML algorithm to carry out this task. 
```

> A supervised ML model is of limited utility if it can only accurately predict phenotype labels in the data it was trained on (i.e., is said to be _overfit_); instead, a model that _generalizes_ or maintains performance when applied to unseen data is of practical benefit.

Reviewer 2: ```suggestion
A supervised ML model is of limited utility if it can only accurately predict phenotype labels in the data it was trained on (this is called _overfitting_); instead, it's more beneficial to develop models that _generalize_ or maintains performance when applied to new data that has not yet been 'seen' by the model.
In later sections, we'll discuss "regularized" models, a strategy for reducing overfitting that can be useful for rare disease datasets. 
```

> Unsupervised ML algorithms can learn patterns or features from unlabeled training data.

Reviewer 2: ```suggestion
In contrast to supervised methods, unsupervised ML algorithms learn patterns or features from **unlabeled** training data.
```

Author: I don't think I'm going to add `In contrast to supervised methods,` in the interest of (ruthlessly) keeping an eye on word count.

Examples of unsupervised learning include principal component analyses (PCA), k-means or hierarchical clustering, or t-distributed stochastic neighbor embedding (t-SNE).

> Prior becoming a part of a classification problem, it is possible that molecular subtype phenotype labels were borne out of the observation that certain sets of samples tended to cluster together when an unsupervised ML approach was applied.

Reviewer 2: I am not entirely sure what this sentence is getting at. I tried to rephrase below: 

```suggestion
For example, in the absence of known molecular subtypes, unsupervised methods like clustering, PCA, or t-SNE have been used to identify tumor subtypes by identifying samples that cluster together. (TODO: CITE)
```

Unsupervised approaches can also extract combinations of features (e.g., genes) that are indicative of a certain cell type or pathway.

> Depending on the method being applied, too few samples can lead to problems identifying signals that are useful for biological discovery or even with the completion of the training process itself.

Reviewer 2: ```suggestion
Often, too few samples (not enough data) leads to challenges in successfully training a model, or in identifying signals that are useful for biological discovery.
```

Though researchers strive to train useful and informative models, there are challenges inherent to applying ML to rare disease datasets. 

> For example, training supervised models requires datasets where the phenotype labels have very little uncertainty (or “label-noise”) [@doi:10.1093/jamia/ocw028] – termed "gold standard" datasets – but rare disease datasets often come with significant label-noise (e.g., _silver standard_ datasets) due to limits in the current understanding of underlying biology.

Reviewer 2: ```suggestion
For example, training supervised models requires datasets where the phenotype labels have very little uncertainty (or “label-noise”) [@doi:10.1093/jamia/ocw028] – termed "gold standard" datasets – but rare disease datasets often come with significant label-noise (e.g., _silver standard_ datasets) due to limits in the current understanding of underlying biology and evolving clinical definitions of many rare diseases. 
```
